ci-operator/step-registry/telcov10n/sno-spoke/gitops/setup/telcov10n-sno-spoke-gitops-setup-commands.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+echo "************ telcov10n Fix user IDs in a container ************"
+[ -e "${HOME}/fix_uid.sh" ] && "${HOME}/fix_uid.sh" || echo "${HOME}/fix_uid.sh was not found" >&2
+
+function set_hub_cluster_kubeconfig {
+  echo "************ telcov10n Get Hub kubeconfig from \${SHARED_DIR}/hub-kubeconfig location ************"
+  export KUBECONFIG="${SHARED_DIR}/hub-kubeconfig"
```

Comment:
I wonder if we can put this env var in the image, since this function is being implemented many times. 
what do you think?

Reply:
I understand your point about reusability, but I think doing what you propose might make it harder to understand what the script does. The question would be, where do you add that code? In each ENV stanza? In that case you are just moving the "problem" around, while making it harder to understand what the script does. Even when debugging, having everything in one place gives you the opportunity to run the script without having to do a lot of tweaking.

Another approach would be, for instances, to move this code into eco-ci-cd branch and source from there. But again, you're moving "code for Prow jobs" to the branch you're supposedly using for "development". If you make changes to your repo, that implies you make changes to all your associated jobs. And I'm not saying that's wrong or not. I'm talking about whether it's the philosophy being followed in Prow. In fact, I have tried to follow the same approach for the rest of the Prow jobs I have checked out that basically consists of being auto-contained.

Reply:
I see, makes sense, thanks!